compiler/semcall.nim

```diff
@@ -232,20 +232,20 @@ proc presentFailedCandidates(c: PContext, n: PNode, errors: CandidateErrors):
         candidates.add("\n  required type for " & nameParam &  ": ")
         candidates.add typeToString(wanted)
         candidates.addDeclaredLocMaybe(c.config, wanted)
-        candidates.add "\n  but expression '"
+        candidates.add "\n  but expression ''"
         if err.firstMismatch.kind == kVarNeeded:
           candidates.add renderNotLValue(nArg)
-          candidates.add "' is immutable, not 'var'"
+          candidates.add "'' 'is immutable, not' ''var''"
         else:
           candidates.add renderTree(nArg)
-          candidates.add "' is of type: "
+          candidates.add "'' 'is of type:' ''"
```

Comment:
instead of concatenating half quoted expressions, use `renderer.quoteExpr` which I had added some time ago in preparation for something like this (feel free to move it if needed).

It can be made more robust by escaping quotes that it finds in the input, and then your `colorError` can correctly handle such escaped quotes to avoid getting confused by it.

Let me know if this needs more clarification!